PhysicsTools/NanoAOD/plugins/TriggerOutputBranches.cc

```diff
@@ -0,0 +1,86 @@
+#include "PhysicsTools/NanoAOD/plugins/TriggerOutputBranches.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/Registry.h"
+
+#include <iostream>
+
+void 
+TriggerOutputBranches::updateTriggerNames(TTree & tree, const edm::TriggerNames & names, const edm::TriggerResults & triggers) 
+{  
+   std::vector<std::string> newNames(triggers.getTriggerNames());
+   if(newNames.empty()) {
+       for(unsigned int j=0;j<triggers.size();j++) {
+ 	   newNames.push_back(names.triggerName(j));
+       }
+   }
+
+   for(auto & existing : m_triggerBranches)
+   {
+       existing.idx=-1;// reset all triggers as not found
+       for(unsigned int j=0;j<newNames.size();j++) {
+	  if(newNames[j]==existing.name) existing.idx=j;		
+       }
+   }
+   // Find new ones
+   for(unsigned int j=0;j<newNames.size();j++) {
+       std::string name=newNames[j];
```

Comment:
should be a const &?

Reply:
better, a range based loop

Reply:
will use `const &`  ( the index is used some lines below )

Reply:
actually it will stay as is since also the string is modified later.